compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -760,7 +760,21 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         };
 
         let trait_ref = self.closure_trait_ref_unnormalized(obligation, substs);
-        let mut nested = self.confirm_poly_trait_refs(obligation, trait_ref)?;
+
+        // FIXME(non_lifetime_binders): Perform the equivalent of a "leak check" here.
```

Comment:
Alternatively, we could let this succeed, then error out during HIR typeck or something, though it would be harder to point to the source of the error.